mock/mock.spec

```diff
@@ -17,7 +16,8 @@ License: GPL-2.0-or-later
 # cd mock
 # git reset --hard %%{name}-%%{version}
 # tito build --tgz
-Source: %{name}-%{version}.tar.gz
+Source0: %{name}-%{version}.tar.gz
+Source1: https://raw.githubusercontent.com/rpm-software-management/mock/main/mock-core-configs/mock.conf
```

Comment:
???

Reply:
This is the way to address the file in %pre when macros are evaluated (before the tarball is extrected).

Reply:
I think you are confusing `%pre` and `%prep`.

Reply:
No no ... the %sysusers_create_compat macro is evaluated at the time when the tarball isn't extracted, yet.  See how it is defined, it contains the shell-script macro `%()`.

Reply:
So this is bad?
https://src.fedoraproject.org/rpms/bodhi-server/blob/rawhide/f/bodhi-server.spec#_125

Reply:
According to my experiments, ... yes?  But maybe I'm doing something wrong.

Reply:
Yeah, I tried to install bodhi-server in fedora:rawhide container, and there's no `bodhi` user/group:
```
[root@db5467446600 /]# grep bodhi /etc/group /etc/passwd
[root@db5467446600 /]# 
```
The reason nobody noticed is that they (probably) install it on servers with systemd.

Reply:
they also try to work around this by defining the file directly in spec
https://src.fedoraproject.org/rpms/bodhi-server/blob/rawhide/f/bodhi-server.spec#_84

it seems like hard task to push the .sysusers file from upstream to `%pre`...

Reply:
That is wrong approach, the -f file needs to contain *shell script*, not the sysusersdir content.

Reply:
Actually, as I realized in the other thread, the feature is now RPM built-in. But it still doesn't work for the bodhi-server :-) confusing as hell.  I think the problem actually is that the dropin has `.sysusers` suffix instead of `.conf`.  I'm trying to propose [a fix here](https://src.fedoraproject.org/rpms/bodhi-server/pull-request/3).